### PR TITLE
feat: allow non-ccc pipelines

### PR DESCRIPTION
**Pull request by omerfirmak**
## 1. Purpose or design rationale of this PR

follower nodes don't need to run CCC on their pending blocks.

## 2. PR title

Your PR title must follow [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) (as we are doing squash merge for each PR), so it must start with one of the following [types](https://github.com/angular/angular/blob/22b96b9/CONTRIBUTING.md#type):

- [ ] build: Changes that affect the build system or external dependencies (example scopes: yarn, eslint, typescript)
- [ ] ci: Changes to our CI configuration files and scripts (example scopes: vercel, github, cypress)
- [ ] docs: Documentation-only changes
- [x] feat: A new feature
- [ ] fix: A bug fix
- [ ] perf: A code change that improves performance
- [ ] refactor: A code change that doesn't fix a bug, or add a feature, or improves performance
- [ ] style: Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc)
- [ ] test: Adding missing tests or correcting existing tests

## 3. Deployment tag versioning

Has the version in `params/version.go` been updated?

- [ ] This PR doesn't involve a new deployment, git tag, docker image tag, and it doesn't affect traces
- [x] Yes

## 4. Breaking change label

Does this PR have the `breaking-change` label?

- [x] This PR is not a breaking change
- [ ] Yes

### CODE REVIEW

File: miner/scroll_worker.go
Comment (Thegaram) [inline miner/scroll_worker.go:521]: `isRunning` will only be true on `signer-0` and `signer-1`, correct? And follower nodes (like `internal-1`) will run ccc in the block validation part.

File: miner/scroll_worker.go
Comment (Thegaram) [inline miner/scroll_worker.go:589]: Why do we need this? So that we don't wait for the timeout?

**Review by Thegaram — DISMISSED** [review]

File: miner/scroll_worker.go
Comment (omerfirmak) [reply miner/scroll_worker.go:589]: Yeah, exactly. If we do wait for the timeout, it is likely that the actual block will be gossiped by the sequencer before we can finalize ours.

File: miner/scroll_worker.go
Comment (omerfirmak) [reply miner/scroll_worker.go:521]: `isRunning` will only be true for nodes with `--mine` flag, so whichever signer is active at that time. This is how the original worker behaved as well. `--ccc` flag doesn't matter in block building.

> And follower nodes (like internal-1) will run ccc in the block validation part.

Yeap, CCC in validation step is configured by the `--ccc` flag as it was before.

Comment (Thegaram) [reply miner/scroll_worker.go:521]: Fyi both signers have the `--mine` flag, it's just that one of them is not an authorized Clique signer so it will fail at the sealing step. (There might be transient periods where both are active, when we're "rotating" signers.)

**Review by Thegaram — APPROVED** [review]

File: miner/scroll_worker.go
Comment (omerfirmak) [reply miner/scroll_worker.go:521]: Hmm, I wasn't aware. That might be an issue. Let me see.

Comment (omerfirmak) [reply miner/scroll_worker.go:521]: But it is unrelated to this PR, so we can merge this when we can.

File: miner/scroll_worker.go
Comment (colinlyguo) [inline miner/scroll_worker.go:590]: Why use Stop() here instead of Release()?

File: miner/scroll_worker_test.go
Comment (colinlyguo) [inline miner/scroll_worker_test.go:1090]: What about checking the length of transactions? more detailed than NotEmpty.

**Review by colinlyguo — APPROVED** [review]

File: miner/scroll_worker.go
Comment (omerfirmak) [reply miner/scroll_worker.go:590]: `Release()` might consume the result that pipeline returns. We still want worker to be able to read the result and process.

File: miner/scroll_worker_test.go
Comment (omerfirmak) [reply miner/scroll_worker_test.go:1090]: That doesn't really matter for us, I would rather keep it the way it is to not need a separate round of approvals if you don't mind:)

Comment (colinlyguo) [reply miner/scroll_worker_test.go:1090]: yeah. never mind. resolved.

**Pull request merged**
